src/Workspaces/Core/Portable/Rename/Renamer.RenameDocumentActionInfo.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        /// <summary>
+        /// Information about rename document calls that allows them to be applied as individual actions.
+        /// </summary>
+        public sealed class RenameDocumentActionInfo
+        {
+            internal RenameDocumentActionInfo(ImmutableArray<RenameDocumentAction> actions, Solution solution)
+            {
+                Solution = solution;
+                ApplicableActions = actions;
+            }
+
+            /// <summary>
+            /// The solution that was used to compute the rename actions
+            /// </summary>
+            public Solution Solution { get; }
+
+            /// <summary>
+            /// All applicable actions computed for the action
+            /// </summary>
+            public ImmutableArray<RenameDocumentAction> ApplicableActions { get; }
```

Comment:
when are there multiple actions?

Reply:
Current known that could be multiple (not in current implementation)

Rename document
Sync namespace
Fix header